templates/hooks/queue/sdk_get_attributes_pre_set_output.go.tpl

```diff
@@ -0,0 +1,8 @@
+	// If the QueueName field is empty, populate it with the last part of the queue ARN
+	// This is a workaround for the fact that the QueueName field is required by the
+	// CreateQueue API call, but not by the GetQueueAttributes API call
+	// Use case: adopting an existing queue by queue URL
+	if ko.Spec.QueueName == nil {
+		split := strings.Split(string(tmpARN), ":")
+		ko.Spec.QueueName = &split[len(split)-1]
+	}
```

Comment:
We need to bring the new `arn.Parse` changes to this template, so that the controller is regenerate-able. I see that we don't have the `"github.com/aws/aws-sdk-go/aws/arn"` import in `sdk.go` and for that we can have two solutions:
- Add a `parseARN` function and the necessary import to `hooks.go` - then you can just use `parseARN` in the template
- Add the import to `sdk.go` template in the code-generator repository + something like `_ = strings.ToLower("")` (L42 in case) the import is not used..

Reply:
@A-Hilaly  thanks for review,  I forget to regenerate the code before pushing,  I have refactored with approach 1